### ISSUE TITLE
Sir please make Android 8.1 rom for htc 628 🙏🏽

### ISSUE THREAD

**Issue by Utkarsh898** [McPrapor/kernel_misu#5]
Sir I'm requesting you please make Android 8.1 rom I'm waiting for years but no Android 8 rom for htc 628 🙏🏽🙏🏽🥺

**Comment by Yeshkumar989** [comment]
I have also htc 628 @McPrapor please make Android 8 rom

**Comment by Yeshkumar989** [comment]
Please sir make Android 8 or 9 rom please 🙏🏽🙏🏽🙏🏽

**Comment by McPrapor** [comment]
I'm sorry, this not something I'm capable of doing. I tried several times, there is some problem with cameras in all my Android 8 build and outdated modem firmware doesn't allow to build anything newer android 8.
Anyway I suggest you to search htc 628 topic on 4pda. There is a madOS rom with this kernel, which is based on android 8. I used it for some years, it's more or less fine as I remember.

**Comment by Yeshkumar989** [comment]
Thanks for replying 🙂🙏🏽

**Comment by Utkarsh898** [comment]
sir i can't find htc 628 madOS i don't understand Russia please can you
provide me link or other way to download rom 🙏🏽

**Comment by McPrapor** [comment]
https://mega.nz/#F!2OoF2SiZ!UvlAgH-C2F3CYJYeJqGzlg
So basically unlock the device, download opengapps and save to the sdcard,
install the recovery package from link above, boot into the new
recovery(it's twrp), make backup of all your partitions(especially nvram
and nvdata), clean caches and data, install the mados package from link
above, install gapps package you downloaded before.
There is a separate boot package which have slightly newer
kernel(boot_1dad568), should better handle charging, could be installed
anytime later without factory reset or cleaning caches.

This is so called "ported" firmware, originally built for coolpad note 3 by
mados team, I took their kernel sources and made it work with htc 628, but
I failed to build the entire rom myself, camera refused to work. Anyway
this one works more or less fine, just remember, in case of any bootloops
first of all go to twrp and restore nvram and nvdata.
